Tools/autotest/sim_vehicle.py

```diff
@@ -231,6 +231,7 @@ def kill_tasks():
             'AntennaTracker.elf',
             'JSBSIm.exe',
             'MAVProxy.exe',
+            'mavproxy.py',
```

Comment:
This is kind of nasty; it's going to kill every mavproxy on the system, on all platforms.  I'm often running several MAVProxys and I'd prefer them not all be killed when we run through here.

I do acknowledge it seems to have this behaviour under Windows :-)

Reply:
And it doesn't fix the issue of not closing the map and console windows !

Reply:
@peterbarker
I agree that this is nasty, but it's basically how tasks were killed before. I think an apt solution is to populate a global set with PIDs of each process as they are spawned. Upon exit, that set would be used to kill processes as opposed to a set of names.

@khancyr
Strange, it works on my end. Are you on Linux?

Reply:
yep, I am Ubuntu 19.10. that doesn't kill all mavproxy process

Reply:
ok I find the error, I don't have psutil ...

Reply:
About why the remain windows aren't closed, it seems that the run_in_terminal command is creating a fork for the mavproxy instance.
That is annoying as we don't have a way to get the pid to kill them